src/EduManagementLab.EfRepository/DataContext.cs

```diff
@@ -8,9 +8,14 @@ public class DataContext : DbContext
         public DataContext(DbContextOptions<DataContext> options) : base(options)
         {
         }
-        
+
         public DbSet<User> Users { get; set; }
         public DbSet<Course> Courses { get; set; }
         public DbSet<Course.Membership> CourseMemberships { get; set; }
+        //Using Entity Ignore self referencing loop
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```

Comment:
Do we need this?

Reply:
I Think we need it when we call GetCourse and include the memberships, it's going to return Course object and include Memberships. And inside Membership object has also Course object  so it's a self referencsin loop. So this method is going to ignore course reference inside course.Membership Object